### DISCUSSION TITLE
Inject handler - Critera issue

### DISCUSSION THREAD

**Discussion by Julian1993ARG** [serenity-is/Serenity#7261] [Help] [answered]
Hi, I’m having an issue when using handlers through dependency injection. When I try to include a base criteria in the request, I get the error: 'T0.[EntityFld] is not a valid field name!'. The error occurs when executing the 'ApplyCriteria' method. I can use 'EqualityFilter' and then filter the data with a 'Where', but that’s not what I want. Also, I don't understand why 'Criteria' gives an error in this case. Could you please help me understand why this is happening?

```
      var fld = MyRow.Fields;
      BaseCriteria crt = new Criteria(fld.EntityFld).In(4,5,6,7);

      List<MyRow> ResponseList = hadler.List(connection, new ListRequest
      {
          Criteria = crt,
          IncludeColumns = new HashSet<string> { "columnName" }
      }).Entities;
```

**Comment by Julian1993ARG** [comment, marked as answer]
My bad.

```
  var fld = MyRow.Fields;
      BaseCriteria crt = new Criteria(fld.EntityFld.Name).In(4,5,6,7);

      List<MyRow> ResponseList = hadler.List(connection, new ListRequest
      {
          Criteria = crt,
          IncludeColumns = new HashSet<string> { "columnName" }
      }).Entities;
```

**Reply by dfaruque** [reply]
Following should work aslo
`var crt = fld.EntityFld.In(4,5,6,7);`
